src/streamlink/plugins/dlive.py

```diff
@@ -0,0 +1,390 @@
+"""
+Plugin for dlive.tv
+"""
+import re
+
+from streamlink.plugin import Plugin
+from streamlink.plugin.api import validate
+from streamlink.stream import HLSStream
+from streamlink.utils import parse_json
+from streamlink import NoStreamsError
+
+
+QUALITY_WEIGHTS = {
+   "src": 1080,
+}
+
+
+class dlive(Plugin):
+    _url_re = re.compile(r"https?://(?:www\.)?dlive\.tv/")
+
+    _INFO_SCHEMA = validate.Schema({
+                                    "data": {
+                                        "userByDisplayName": {
+                                            "username": validate.text
+                                        }
+                                    }
+                                }
+                            )
+
+    @classmethod
+    def can_handle_url(cls, url):
+        return cls._url_re.match(url)
+
+    @classmethod
+    def stream_weight(cls, key):
+        weight = QUALITY_WEIGHTS.get(key)
+        if weight:
+            return weight, "dlive"
+
+        return Plugin.stream_weight(key)
+
+    def _get_streams(self):
+        reqUrl = "https://graphigo.prd.dlive.tv/"
```

Comment:
put this url into the class so you can use it with `self.API_URL` or something similar